// node-tide/dist/transformers/account.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var moment = require("moment");
var transformer_1 = require("../transformers/transformer");
var Account = (function (_super) {
    __extends(Account, _super);
    function Account() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Account.prototype.item = function (item) {
        return Promise.resolve(item)
            .then(function (item) {
            return {
                id: item.accountId,
                companyId: item.companyId,
                accountNumber: item.accountNumber,
                sortCode: item.sortCode,
                name: item.name,
                balance: item.balance,
                availableBalance: item.availableBalance,
                currencyCode: item.currencyCode,
                createdOn: moment(item.isoCreatedOn),
                updatedOn: moment(item.isoUpdatedOn)
            };
        });
    };
    return Account;
}(transformer_1.Transformer));
exports.Account = Account;
