// node-tide/dist/transformers/company.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var moment = require("moment");
var transformer_1 = require("../transformers/transformer");
var Company = (function (_super) {
    __extends(Company, _super);
    function Company() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Company.prototype.item = function (item) {
        return Promise.resolve(item)
            .then(function (item) {
            return {
                id: item.companyId,
                number: item.number,
                name: item.name,
                sicCode: item.sicCode,
                vatNumber: item.vatNumber,
                tradingName: item.tradingName,
                registered: item.registered,
                createdOn: moment(item.isoCreatedOn),
                updatedOn: moment(item.isoUpdatedOn)
            };
        });
    };
    return Company;
}(transformer_1.Transformer));
exports.Company = Company;
