// node-tide/dist/transformers/transaction.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = Object.setPrototypeOf ||
        ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
        function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
var moment = require("moment");
var transformer_1 = require("../transformers/transformer");
var Transaction = (function (_super) {
    __extends(Transaction, _super);
    function Transaction() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    Transaction.prototype.item = function (item) {
        return Promise.resolve(item)
            .then(function (item) {
            return {
                id: item.transactionId,
                accountId: item.accountId,
                amount: item.amount,
                type: item.type,
                transactionRef: item.txnRef,
                transactionOn: moment(item.isoTransactionDateTime),
                appliedOn: moment(item.isoAppliedDateTime),
                clearedOn: (item.isoClearedDateTime),
                maskedPan: item.maskedPan,
                status: item.status,
                description: item.description,
                createdOn: moment(item.isoCreatedOn),
                updatedOn: moment(item.isoUpdatedOn),
                category: {
                    id: item.categoryId,
                    name: item.categoryName,
                    type: item.categoryType
                }
            };
        });
    };
    return Transaction;
}(transformer_1.Transformer));
exports.Transaction = Transaction;
