// node-tide/dist/services/service.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var request = require("request-promise-native");
var url = "https://api.tide.co/tide-backend/rest/api/v1";
var Service = (function () {
    function Service() {
        this.accessToken = undefined;
    }
    Service.prototype.setAccessToken = function (accessToken) {
        this.accessToken = accessToken;
        return this;
    };
    Service.prototype.postRequest = function (path, data) {
        return this.performRequest('POST', path, data);
    };
    Service.prototype.getRequest = function (path, data) {
        return this.performRequest('GET', path, data);
    };
    Service.prototype.performRequest = function (type, path, data) {
        var _this = this;
        return Promise
            .resolve({ endpoint: "" + url + path, data: data })
            .then(function (_a) {
            var endpoint = _a.endpoint, data = _a.data;
            return request({
                method: type,
                uri: endpoint,
                headers: _this.accessToken === undefined ? {} : {
                    Authorization: "Bearer " + _this.accessToken
                },
                body: data,
                json: true
            });
        });
    };
    return Service;
}());
exports.Service = Service;
