// node-tide/dist/index.js
"use strict";
function __export(m) {
    for (var p in m) if (!exports.hasOwnProperty(p)) exports[p] = m[p];
}
Object.defineProperty(exports, "__esModule", { value: true });
__export(require("./services"));
__export(require("./entities"));
var Services = require("./services");
var Tide = (function () {
    function Tide(accessToken) {
        if (accessToken === void 0) { accessToken = undefined; }
        this.auth = (new Services.Auth());
        if (accessToken !== undefined) {
            this.setAccessToken(accessToken);
        }
    }
    Tide.prototype.setAccessToken = function (accessToken) {
        if (accessToken === void 0) { accessToken = undefined; }
        if (accessToken === undefined) {
            throw new Error("Access Token not Supplied");
        }
        this.companies = (new Services.Companies()).setAccessToken(accessToken);
        this.accounts = (new Services.Accounts()).setAccessToken(accessToken);
        this.transactions = (new Services.Transactions()).setAccessToken(accessToken);
    };
    return Tide;
}());
exports.Tide = Tide;
